src/main/java/org/folio/rest/impl/PutOrderLineByIdHelper.java

```diff
@@ -218,6 +219,11 @@ public boolean isUpdateSuccessful() {
                .filter(id -> !existingItemIds.contains(id))
                .forEach(itemId -> futuresList.add(createPiece(poLineId, itemId)));
 
+        // Create pieces when item record does not exists
```

Comment:
The `createPieces` method will be invoked only if any inventory updates required. Please add additional conditions in the begining of the `updateInventory` method to cover this case.

Reply:
Currently, only `updateInventory` is invoking `createPieces` so why do we need a conditional check? No one else is calling `createPieces`?

Reply:
I mean this check:
```java
    int expectedItemsQuantity = calculateInventoryItemsQuantity(compPOL);
    if (expectedItemsQuantity == 0) {
      logger.debug("PO Line with '{}' id does not require inventory updates", compPOL.getId());
      return completedFuture(null);
    }
```

If nothing required to be created in inventory, the `createPieces`will not be called at all.